src/foregroundTimeTracker.ts

```diff
@@ -0,0 +1,100 @@
+export default class ForegroundTimeTracker {
+    private isTrackerActive: boolean = false;
+    private localStorageName: string = '';
+    public startTime: number = 0;
+    public totalTime: number = 0;
+
+    constructor(timerKey: string) {
+        this.localStorageName = `mp-time-${timerKey}`;
+        this.loadTimeFromStorage();
+        this.handleVisibilityChange = this.handleVisibilityChange.bind(this);
+        this.syncAcrossTabs = this.syncAcrossTabs.bind(this);
+        this.init();
```

Comment:
can this be 2 different methods for organizational purposes?  1 for setting up the localStorage (and perhaps pass localstorageName to loadTimeFromStorage), then another for all of the event handlers.